src/NUnitFramework/tests/Internal/Execution/ParallelWorkItemDispatcherTests.cs

```diff
@@ -0,0 +1,79 @@
+﻿// ***********************************************************************
+// Copyright (c) 2017 Charlie Poole
+//
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+//
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+// ***********************************************************************
+
+#if PARALLEL
+using System.Collections.Generic;
+using System.Linq;
+
+namespace NUnit.Framework.Internal.Execution
+{
+    public class ParallelWorkItemDispatcherTests
+    {
+        private const int LEVEL_OF_PARALLELISM = 4;
+        private ParallelWorkItemDispatcher _dispatcher;
+
+        [SetUp]
+        public void CreateDispatcher()
+        {
+            _dispatcher = new ParallelWorkItemDispatcher(LEVEL_OF_PARALLELISM);
+        }
+
+        [Test]
+        public void ConstructorCreatesQueues()
+        {
+            var queues = new List<WorkItemQueue>(_dispatcher.Queues);
+
+            Assert.That(List.Map(queues).Property("Name"), 
```

Comment:
Minor, but assuming we have LINQ, these two lines could be simplified down to `_dispatcher.Queues.Select(q => q.Name)` and then it would be refactor safe.